source_code/main_mcu/src/LOGIC/logic_fido2.c

```diff
@@ -154,7 +154,16 @@ void logic_fido2_process_exclude_list_item(fido2_auth_cred_req_message_t* reques
         return;
     }
     else
-    {        
+    {
+        /* Wait for user ACK */
+        cust_char_t* display_cred_prompt_text;
+        custom_fs_get_string_from_file(CRED_ALREAD_PRESENT_TEXT_ID, &display_cred_prompt_text, TRUE);
+        confirmationText_t prompt_object = {.lines[0] = display_cred_prompt_text};
+        gui_prompts_ask_for_confirmation(1, &prompt_object, FALSE, TRUE, FALSE);
```

Comment:
hmmm there's an issue with calling gui_prompts_ask_for_confirmation, as it will re-use the RX buffer you got the FIDO2 request from. therefore, request->cred_ID that you're using just after will get corrupted if (for example) the user disconnects his device.
So just make a copy of it and you should be good